### DISCUSSION TITLE
Envied Obfuscate

### DISCUSSION THREAD

**Discussion by OmerDemirkiran** [VB10/architecture_template_v2#29] [Q&A]
Hocam merhaba,
Flutter Mimari v2b5 videosunda anlattığınız Envied ile ortam değişkenlerini yönetme işlemlerini izledim ve denedim. Anlatımınızda .env dosyaları ile özel bilgileri proje içine dahil etmeyerek daha güvenli bir şekilde saklandığından bahsetmişsiniz. Envied paketi .env dosyasındaki verileri xyz.g.dart dosyası ile yine uygulama içinde değişken olarak saklıyor. Obfuscate seçeneğini açtığımızda verileri şifreleyerek daha güvenli olduğu belirtiliyor ama kullanılan şifreleme algoritması çok basit bir algoritma. Oluşturulan şifrelenmiş xyz.g.dart dosyasındaki verileri incelediğinizde şifrelenen veri için iki adet int türünde liste görüyoruz. Bu listedeki değerlere xor işlemi yaptığımızda sonuç karakterin ASCII kodu oluyor, bunu da String'e dönüştürüp birleştirdiğimizde bizim sakladığımız verimiz ortaya çıkıyor. Şifrelenmiş xyz.g.dart dosyasındaki değişkeni döndüren fonksiyonu incelerseniz listeleri map'leyerek xor işlemi yapıp çıkan sonuçları birleştirerek bize döndürüyor. Bu yöntemi kullanmak ile verileri direkt kendi oluşturduğumuz bir ConstantString sınıfında tutmak arasında bir fark yok gibi görünüyor. Uygulamanın kaynak koduna erişebilen biri basit bir xor işlemi ile sakladığımız verileri yine görebiliyor. Envied paketi ve .env dosyası kullanmanın bir katkısı yok gibi görünüyor. Bu konuda ne düşünüyorsunuz? Daha önceki eğitimlerinizde kullandığınız flutter_dotenv paketi Envied paketi gibi uygulama içinde .g.dart dosyası oluşturmadan .env dosyasındaki verileri okuyor, bu daha güvenli bir yaklaşım mı yoksa .env dosyasını pubspec.yaml içinde assets bölümüne girdiğimiz için yine kaynak koda dahil ediliyor ve verileri normal bir json dosyasında saklamaktan farksız hale mi geliyor?

**Comment by alihansoykal** [comment]
Dostum selam bu durumla alakali bir podcast dinlemistim. Bir bak istersen. [link](https://open.spotify.com/episode/3Hj7cPnPJdJYGrmhilAXc3?si=35cb56661fe44693)
